pvlib/pvsystem.py

```diff
@@ -1133,6 +1121,28 @@ def calcparams_desoto(effective_irradiance, temp_cell,
          Source: [4]
     '''
 
+    # test for use of function pre-v0.6.0 API change
+    if isinstance(a_ref, dict) or \
+       (isinstance(a_ref, pd.Series) and ('a_ref' in a_ref.keys())):
```

Comment:
I don't understand the grouping of statements here. I would have thought you'd group on the ``or`` rather than the ``and``. In any case, I think ``isinstance(a_ref, (dict, pd.Series))`` alone may be what we want. A dict or Series that does not have 'a_ref' will fail in the equations that follow. Better to have it fail in the key look up. But maybe I am just misunderstanding.

Reply:
Tested the `if` block on my own.

I added that block because I am assuming a use case where multiple sets of module parameters are passed into the new function as `pandas.Series` objects, and I didn't want the new function to fail in that case.